### PR TITLE
[Improve] Improve Jdbc connector error message when datatype unsupported

### PR DESCRIPTION
**Pull request by Hisoka-X**
### Purpose of this pull request
This PR improve the error message when some datatype SeaTunnel doesn't supported. Because Jdbc support multi-table now, we should return all unsupported datatype in all select table in one error message.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
add new test.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/utils/CatalogUtils.java [hunk 1]
```diff
@@ -201,7 +205,7 @@ public static CatalogTable getCatalogTable(Connection connection, TablePath tabl
                 catalogName);
     }
 
-    public static CatalogTable getCatalogTable(ResultSetMetaData resultSetMetaData)
+    public static CatalogTable getCatalogTable(ResultSetMetaData resultSetMetaData, String sqlQuery)
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/utils/CatalogUtils.java:208]: Incoming parameters should be used for logical processing

```java
public static CatalogTable [your call method](ResultSetMetaData resultSetMetaData, String sqlQuery) {
    try {
        return getCatalogTable(resultSetMetaData);
    } catch (Exception e) {
        throw CommonError.getCatalogTableWithUnsupportedType(..., sqlQuery);
    }
}

public static CatalogTable getCatalogTable(ResultSetMetaData resultSetMetaData) {
    ...
}
```

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/utils/CatalogUtils.java:208]: I think error handling is also a very important part of the logic. If we don't do this, we need to perform try catch everywhere where this method is called, and then require the caller to return appropriate error information, which is very unreasonable.

Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/utils/CatalogUtils.java:208]: 👌

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**